src/polycap-out.c

```diff
@@ -8,15 +8,19 @@ void polycap_out(size_t n_energies, double *energies, double *efficiencies)
 {
 	hid_t file, dataset;
 	hid_t datatype, dataspace; //handles
+	hsize_t n_energies_temp;
 	herr_t status;
-	char h5filename = "polycap_out.h5"
+	char h5filename[] = "polycap_out.h5";
+
 
 	//Create new HDF5 file using H5F_ACC_TRUNC and default creation and access properties
 	file = H5Fcreate(h5filename, H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT); 
 
 	//Write energies array
+	//Define temporary dataset dimension
+	n_energies_temp = n_energies;
 	//Describe size of the array and make fixed data space
-	dataspace = H5Screate_simple(1, n_energies, NULL); //RANK = 1 due to 1D dataset
+	dataspace = H5Screate_simple(1, &n_energies_temp, NULL); //RANK = 1 due to 1D dataset
 	//Define datatype for the dataspace
 	datatype = H5Tcopy(H5T_NATIVE_DOUBLE); //dataset consists of doubles
```

Comment:
volgens mij moet je die `datatype` hier niet gebruiken. Gebruik gewoon direct `H5_NATIVE_DOUBLE`. Die little endian storage flag is overbodig aangezien vrijwel elke architectuur dat al is.

Moet je als dataset naam trouwens er geen slash voorzetten zoals `"/Energies"`?